### ISSUE TITLE
ch4/ofi/psm2: psm2_ep_connect intermittently failing for some coll tests 

### ISSUE THREAD

**Issue by sagarth** [pmodels/mpich#5553]
Some collective tests are failing intermittently, during finalize step. Problem reported during a local debug run.

Config: psm2 + multi-vci + SEP (--disable-ofi-domains)

Error:
```
libfabric:34420:psm2:av:psmx2_epid_to_epaddr():230<warn> psm2_ep_connect retured error Operation timed out, remote epid=1f0302.If it is a timeout
```
tests: `./coll/opbxor 5` `./coll/bcastzerotype 5 MPIR_CVAR_BCAST_DEVICE_COLLECTIVE=0 MPIR_CVAR_BCAST_INTRA_ALGORITHM=scatter_recursive_doubling_allgather` `./init/session_mult_init 4` etc.

Finding: Problem occurs during test finalize, rank 0 is still at `MPI_Reduce` trying to connect to a remote ep to receive data whereas other ranks have send their data and proceeded to `MPI_Finalize`. Rank 0 tries to connect to an already closed ep 

Logs from test bcastzerotype:
```
[4] libfabric:62990:psm2:core:psmx2_trx_ctxt_free():190<info> epid: 0000000000040e02 (tx+rx)
[0] libfabric:62986:psm2:av:psmx2_epid_to_epaddr():230<warn> psm2_ep_connect retured error Operation timed out, remote ep
```

**Comment by hzhou** [comment]
@sagarth Can you reproduce the issue without multiple VCI? We need to determine whether the issue is vci specific.

**Comment by sagarth** [comment]
@hzhou Actually yes. It just replicated for single-vci too.

**Comment by hzhou** [comment]
Does it depend on SEP?

**Comment by sagarth** [comment]
Yes, SEP dependent.

**Comment by hzhou** [comment]
With single vci, what is the difference between SEP and ofi domain?

**Comment by sagarth** [comment]
It seems one class of difference is the flags that are passed to ofi. But let me take back the statement about SEP. I'm not fully sure if I've tested it with multiple-domains.

**Comment by hzhou** [comment]
To clarify, since it can be reproduced with single vci, there is nothing to do with multiple-domains, right?

**Comment by sagarth** [comment]
Right, problem has been replicated with a single domain and SEP enabled. Haven't tested with SEP disabled.

**Comment by hzhou** [comment]
@sagarth Do you still see these failures? I suspect this is of the same nature as https://github.com/pmodels/mpich/issues/5975. Could you try add `psm2` to the following branch and see if it fixes the issue? --https://github.com/pmodels/mpich/blob/36aaebbf6db2bb5a15a5ea8b7e18422f12d61e18/src/mpid/ch4/netmod/ofi/ofi_init.c#L864-L869

**Comment by sagarth** [comment]
> @sagarth Do you still see these failures? I suspect this is of the same nature as #5975. Could you try add `psm2` to the following branch and see if it fixes the issue? --
> 
> https://github.com/pmodels/mpich/blob/36aaebbf6db2bb5a15a5ea8b7e18422f12d61e18/src/mpid/ch4/netmod/ofi/ofi_init.c#L864-L869

@hzhou In our internal code, we have been using this barrier for psm2. The barrier has helped resolve this problem.

**Comment by hzhou** [comment]
Ahh, then we should apply the fix to the main branch.
